include/exadg/grid/grid_utilities.h

```diff
@@ -245,6 +246,52 @@ create_fine_and_coarse_triangulations(
   }
 }
 
+/**
+ * This function reads an external triangulation. The path to the triangulation file should
+ * be defined in the input file and then stored in GridData. This function extracts it from
+ * GridData and stores it in the triangulation "tria".
```

Comment:
I am still not happy with this comment (and the ones below). Note that this function is not aware of any "input files". I can use this function in ExaDG without any input file! Hence, we are not allowed to refer to an input file here in my opinion.

Reply:
Yes, I agree that this function is independent from the input file. However, I think that we should explain somewhere, how this function was meant to be used. And I thought this would be a good location for this. Otherwise, the user might not know the possibility of defining the path in the input_file at all.

Reply:
Yes, I see your intention. But this is a clear indication that the design is not good enough right now. Imagine someone forgot to add `grid_parameters.add_parameters()` in `Structure::ApplicationBase`. One will then run into this assert, but adding the line with `file_name` to the input file will not help, because it is kind of a bug in the software (not being implemented so far) and the parameter will never be read from the input file. The error message of the above assert just causes more confusion. A function and its documentation in `grid/grid_utilities.h` has to work for all modules/applications in ExaDG, not just Poisson where this have already been implemented. But also for Poisson, the implementation might be changed in the future. How can you guarantee that the developer doing these changes will also adapt `grid_utilities.h`?

Reply:
I understand your point better now! I think extending this comment with the whole procedure of reading a grid_file from the input_file would be too much here. I will remove input_file from this comment and adapt the wording accordingly.